src/Microsoft.TestPlatform.CommunicationUtilities/Serialization/TestObjectConverter.cs

```diff
@@ -55,12 +55,17 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
         // key value pairs.
         foreach (var property in properties)
         {
-            var testProperty = property["Key"].ToObject<TestProperty>(serializer);
+            var testProperty = property["Key"]?.ToObject<TestProperty>(serializer);
```

Comment:
On `TestCaseConverter` you are doing `property?["Key"]` do we need the same here?

Reply:
I think doing that confuses the compiler to think that property can be null afterwards, even though we continue when testProperty is null. In any case we are probably not serializing null pair in the dictionary, so this should be safe. It also worked till now.